Added mid_time() to FileMetadata  issue #53

safimuhammad opened the pull request:
- mid_time() is now moved to FileMetadata class, calculated mid-range of file metadata.
- Test case for FileMetadata.mid_time() also added in test_file_metadata.py. 

ps: Thanks for the help, its a great project.

safimuhammad committed:
mid_range function moved to FileMetadata, updated tests also included

safimuhammad committed:
pre-commit ran on edited files

ebuchlin commented:
Thanks for the PR, but as your branch is taken from the develop branch, could you make the target solo-spice:develop instead of solo-spice:main?

ebuchlin commented on sospice/catalog/file_metadata.py:
```diff
@@ -345,6 +345,21 @@ def get_observer(self):
+        pd.Timestamp
+            The middle time of the observation.
+        """
+        if self.metadata["DATE-BEG"] and self.metadata["TELAPSE"]:
```
I don't think that these checks are needed (we can assume these keys to exist, as their presence is checked by `FileMetadata.validate()` (by comparing to `required_columns`)

ebuchlin commented on sospice/catalog/tests/test_file_metadata.py:
```diff
@@ -116,6 +116,13 @@ def test_get_observer(self, file_metadata):
+        fm = file_metadata
+        date_beg = fm.metadata["DATE-BEG"]
+        telapse = fm.metadata["TELAPSE"]
+        expected_mid_time = date_beg + pd.Timedelta(seconds=telapse / 2)
```
Not sure whether the expected mid time should be obtained by the same formula as in the tested function, or it should simply be a constant.

safimuhammad replied:
I think mid_range formula should remain the same, using a constant is a good option too.

I'll add a constant to resolve this

ebuchlin reviewed (commented):
Thanks for the PR (please see my review). Also, as your branch is taken from the develop branch, could you please make the target solo-spice:develop instead of solo-spice:main?

safimuhammad committed:
tests updated, validations removed

safimuhammad commented:
- changed the branch from main to develop.
- removed validations from file_metadata.py.
- updated test_file_mid_time(), added a constant value instead of a formula.
